### PR TITLE
FE-2168 label is not associated with the input

### PR DESCRIPTION
**Pull request by ianforster-sage**
# Description
Passes the id to the Label htmlFor prop so that the label is properly associated with the Input component

- fixed defect with label click not focusing on input
- refactored withUniqueName to withUniqueProp to better reflect its purpose

# TODO
- [x] Release notes

### CODE REVIEW

File: src/utils/helpers/with-unique-prop/with-unique-prop.hoc.js [hunk 1]
```diff
@@ -2,8 +2,10 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import guid from '../guid';
 
-const withUniqueName = (WrappedComponent) => {
-  class WithUniqueName extends React.Component {
+const withUniqueProp = (WrappedComponent) => {
```
Comment (grabkowski) [inline src/utils/helpers/with-unique-prop/with-unique-prop.hoc.js]: shouldn't that be `withUniqueId`?

Comment (damianOfKyan) [reply src/utils/helpers/with-unique-prop/with-unique-prop.hoc.js]: `withUniqueProp` can be expanded in the future. At the moment it offers `id` and `label` `guid()`.

Comment (grabkowski) [reply src/utils/helpers/with-unique-prop/with-unique-prop.hoc.js]: what do you think about `withUniqueIdProps`? `withUniqueProp` is too general in my opinion.

Comment (damianOfKyan) [reply src/utils/helpers/with-unique-prop/with-unique-prop.hoc.js]: Resolved in 608fe850ca58c210e3e62d5af6e4c713dba037b9

**Comment by sianford** [comment]
All of the labels in all of the stories for all input components with labels have the `for=""`attribute. Selecting the label brings the input into focus and lands the cursor in the input or highlights existing content, as required. A screenreader identifies the label as being connected to the field. Testing passed ✅

**Pull request merged**